virttest/test_setup.py

```diff
@@ -1438,6 +1438,17 @@ def check_vfs_count(self):
         expected_count = int((re.findall("(\d+)", self.driver_option)[0])) * len(self.get_pf_ids())
         return (self.get_vfs_count() == expected_count)
 
+    def get_controller_type(self):
+        """
+        Get the Controller Type for SR-IOV Mellanox Adapter PFs
+        """
+        cmd = "lspci | grep 'Mellanox' | awk '/%s/ {print $2}'" % self.pf_filter_re
+        controller_type = process.system_output(cmd, shell=True).strip().split("\n")[-1]
+        if not controller_type:
+            raise exceptions.TestSetupFail("Unable to fetch the controller details : %s"
```

Comment:
API get_controller_type() should probably return None instead of raising setup fail, as it would check for the controller type if available else returns None.
```python
try:
    cmd = "lspci | awk '/%s/ {print $2}'" % self.pf_filter_re
    controller_type = process.system_output(cmd, shell=True).strip().split("\n")[-1]
except IndexError:
    logging.debug("Unable to fetch the controller details")
    return None
```
raising the setup failure should be taken care during actual setup.

Reply:
@balamuruhans Thanks for reviewing, have made the necessary changes.